sample_consensus/include/pcl/sample_consensus/boost.h

```diff
@@ -1,46 +0,0 @@
-/*
- * Software License Agreement (BSD License)
- *
- *  Point Cloud Library (PCL) - www.pointclouds.org
- *  Copyright (c) 2012-, Open Perception, Inc.
- *
- *  All rights reserved.
- *
- *  Redistribution and use in source and binary forms, with or without
- *  modification, are permitted provided that the following conditions
- *  are met:
- *
- *   * Redistributions of source code must retain the above copyright
- *     notice, this list of conditions and the following disclaimer.
- *   * Redistributions in binary form must reproduce the above
- *     copyright notice, this list of conditions and the following
- *     disclaimer in the documentation and/or other materials provided
- *     with the distribution.
- *   * Neither the name of the copyright holder(s) nor the names of its
- *     contributors may be used to endorse or promote products derived
- *     from this software without specific prior written permission.
- *
- *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
- *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
- *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
- *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
- *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
- *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
- *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
- *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
- *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
- *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
- *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
- *  POSSIBILITY OF SUCH DAMAGE.
- *
- * $Id: lmeds.h 1370 2011-06-19 01:06:01Z jspricke $
- *
- */
-
-#pragma once
-
-#if defined __GNUC__
-#  pragma GCC system_header 
-#endif
-
-#include <boost/random.hpp>
```

Comment:
I think we might need to deprecate this header out. If we agree on this, the header's deletion from the CMakeLists.txt needs to be reverted.

Reply:
Any idea how to add deprecation warning here, so it will getting triggered by "-Wdeprecated"? `[[deprecated]]` can't be uses standalone or for `#include`

Reply:
I'd just do the same as we did with FZAPI. Empty this header and put an #error inside.

Reply:
> Any idea how to add deprecation warning here, so it will getting triggered by "-Wdeprecated"? `[[deprecated]]` can't be uses standalone or for `#include`

Not really. Submit a proposal in the std c++ committee :sweat_smile: ?

I would rather follow up with a pragma warning. The idea is to exactly to not "explode" our consumers code.

Reply:
So use `#error` or `#warning` now? First ist solution like we used for FZAPI (will "explode" consumers code), during second doesn't explode consumers code, but may be not seen.

Reply:
From Sergey's reaction it seems he also acceded to the use of `#warning` as well. The warning will be seen under normal circumstances. If the user disabled general warnings, that's a conscious decision on his/her part and there's nothing we can do there. It there's simply no one checking warnings, than it is no different than removing the file, but we did our fair best to let the user know.

Reply:
Fully agree. Here is my explicit "yes" ;)